### PR TITLE
WIP: Upgrade to Mailer from swiftmailer

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Ni
| New feature/enhancement? (use the a.x branch)      | 5.x
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | 5.x
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
First Attempt to move to Symfony mailer instead of Swiftmailer.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10742"></a>

### CODE REVIEW

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
Thanks for signing the CLA @plotnikovas and welcome to the community! 👋

**Comment by escopecz** [comment]
@mabumusa1 please also write the details of breaking changes into https://github.com/mautic/mautic/blob/5.x/UPGRADE-5.0.md. It's easier to do it while you have it fresh in memory than after several months.

**Comment by mabumusa1** [comment]
Hello there,

This is a quick video to show the differences between what we have in Mautic4 with what we will have in Mautic5, we integrated `symfony/mailer` and `symfony/messegner` for better performance and higher sending speed also to keep Mautic using the last supported packages from Symfony.

Here is a quick video of how things will be changed

https://user-images.githubusercontent.com/12627658/156060994-68f28165-27da-4dbb-9211-f33a7ba6a9fe.mp4

**Comment by mabumusa1** [comment]
> @mabumusa1 please also write the details of breaking changes into https://github.com/mautic/mautic/blob/5.x/UPGRADE-5.0.md. It's easier to do it while you have it fresh in memory than after several months.

I will do that for sure, once we mark this PR as ready for testing I will write the instructions

**Comment by escopecz** [comment]
The video is great! One small thing that caught my eye was that the messages in the Redis quque were encoded as serialized array. Can we switch it to JSON as it is more readable, less prone to unserialization errors and more languages can decode it?

File: app/config/config.php [hunk 1]
```diff
@@ -83,6 +83,24 @@
...
+        //'message_bus' => 'email.bus', // The message_bus option was introduced in Symfony 5.1.
+        'dsn'         => '%env(MAUTIC_MAILER_DNS)%',
+    ],
+    'messenger'            => [
+        'default_bus' => 'email.bus',
+        'buses'       => [
+            'email.bus' => null,
+        ],
+        'transports'  => [
+            'email_transport' => [
```
Comment (kuzmany) [inline app/config/config.php:96]: Awesome.

I do not understand options hosts/path under queue > redis configuration.

image


We use setup of messenger something like 

 ```
  'dsn'            => 'amqp://xxx:xxx@redis.local:7777',
        'options'        => [
            'queues'     =>
                [
                    'queue' =>
                        [
                            'binding_keys' =>
                                [
                                    0 => 'xxx',
                                ],
                        ],
                ],
            'auto_setup' => true,
            'exchange'   =>
                [
                    'name'                        => 'email',
                    'type'                        => 'direct',
                    'default_publish_routing_key' => xxx',
                ],
        ]
```

Probaby we should allow modify it by local parameter with retry_strategy too 

```
      'transports'  => [
            'email_transport' => [
                'dsn' => '%env(MAUTIC_MESSENGER_EMAIL_TRANSPORT_DSN)%',
                'options' => $configParameterBag->get('messenger_email_transport_options'),
                'retry_strategy' => $configParameterBag->get('messenger_email_transport_retry_strategy'),
            ],
        ],
```

Comment (pasxel) [reply app/config/config.php:96]: @kuzmany Thank you for feedback.

> I do not understand options hosts/path under queue > redis configuration.

We have implementation https://symfony.com/doc/current/messenger.html#redis-transport for redis. So `host` is host for redis (like `redis.local`). Path is a stream name (default `messages`). Maybe better rename to `Stream name`.

As for me, ideally to have only one DSN string for all transports (Every transport has their own options and configuration). So options are also can be set using DSN. 
Good point, to move `retry_strategy` as additional parameters for messenger.

Comment (kuzmany) [reply app/config/config.php:96]: > As for me, ideally to have only one DSN string for all transports

Yes, that would be unify solution. 

> So options are also can be set using DSN.

That's not true. Special options must be passed by options parameter, like example above.

**Comment by mabumusa1** [comment]
@escopecz @kuzmany right now we can not use the JSON serializer because there is a bug in `Mailer` mentioned here 

https://github.com/symfony/symfony/issues/33394

**Comment by mabumusa1** [comment]
> @mabumusa1 please also write the details of breaking changes into https://github.com/mautic/mautic/blob/5.x/UPGRADE-5.0.md. It's easier to do it while you have it fresh in memory than after several months.

@escopecz I did write this

File: app/config/config.php (hunk 1 above)
Comment (mabumusa1) [reply app/config/config.php:96]: @kuzmany all the ideas you mentioned are implemented in #11180 11180

**Comment by mabumusa1** [comment]
Closing this PR in favor of [#11180](https://github.com/mautic/mautic/pull/11180)

**Pull request closed without merging**